app/services/auto_assignment/assignment_service.rb

```diff
@@ -0,0 +1,99 @@
+class AutoAssignment::AssignmentService
+  pattr_initialize [:inbox!]
+
+  def perform_for_conversation(conversation)
+    return false unless assignable?(conversation)
+
+    agent = find_available_agent
+    return false unless agent
+
+    assign_conversation(conversation, agent)
+  end
+
+  def perform_bulk_assignment(limit: 100)
+    return 0 unless inbox.enable_auto_assignment?
+
+    assigned_count = 0
+
+    unassigned_conversations(limit).each do |conversation|
+      assigned_count += 1 if perform_for_conversation(conversation)
+    end
+
+    assigned_count
+  end
+
+  private
+
+  def assignable?(conversation)
+    inbox.enable_auto_assignment? &&
+      conversation.status == 'open' &&
+      conversation.assignee_id.nil?
+  end
```

Comment:
I'm not sure about this condition. Why do we need this? 

The `inbox.enable_auto_assignment?` is already checked in the perform_bulk_assignment method. We are taking the unassigned open conversations. So, why do we need to check this again when assigning?